### PR TITLE
Get a config from toml and create a network

### LINKED ISSUE #97: Configuration toml to network
Reverse the dump_to_toml function in order tto create form a toml file networks;

### PR DESCRIPTION
**Pull request by wirednkod**
Fixes #97

### CODE REVIEW

**Review by l0r1s — CHANGES_REQUESTED** [review]

File: crates/configuration/src/network.rs [hunk 1]
```diff
@@ -178,6 +180,97 @@ impl NetworkConfigBuilder<Initial> {
         Self::default()
     }
 
+    pub fn toml_to_network(path: &str) -> Result<NetworkConfig, toml::ser::Error> {
```
[the remark this answers is not in the document]
Comment (pepoviola) [reply crates/configuration/src/network.rs:183]: Agree, I think moving to `NetworkConfig` and call it `load_from_toml` would be nice. :)

Comment (pepoviola) [inline crates/configuration/src/network.rs:183]: We should also need some specialized Err (or we can use anyhow at the moment).

File: crates/configuration/src/network.rs [hunk 2]
```diff
@@ -178,6 +180,97 @@ impl NetworkConfigBuilder<Initial> {
...
+        let file_str = fs::read_to_string(path).unwrap();
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:184]: `unwrap` will panic

File: crates/configuration/src/relaychain.rs [hunk 3]
```diff
@@ -12,20 +12,20 @@ use crate::shared::{
...
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct RelaychainConfig {
     chain: Chain,
     default_command: Option<Command>,
     default_image: Option<Image>,
     default_resources: Option<Resources>,
     default_db_snapshot: Option<AssetLocation>,
     #[serde(skip_serializing_if = "std::vec::Vec::is_empty")]
-    default_args: Vec<Arg>,
+    pub(crate) default_args: Vec<Arg>,
```
Comment (pepoviola) [inline crates/configuration/src/relaychain.rs:23]: Do we need this to have access from the `crate`?

File: crates/configuration/src/relaychain.rs [hunk 4]
```diff
@@ -12,20 +12,20 @@ use crate::shared::{
...
     chain_spec_path: Option<AssetLocation>,
     random_nominators_count: Option<u32>,
     max_nominations: Option<u8>,
     #[serde(skip_serializing_if = "std::vec::Vec::is_empty")]
-    nodes: Vec<NodeConfig>,
+    pub(crate) nodes: Vec<NodeConfig>,
```
Comment (pepoviola) [inline crates/configuration/src/relaychain.rs:28]: Maybe we can replace this with a method like `set_nodes`

File: crates/configuration/src/network.rs [hunk 5]
```diff
@@ -178,6 +180,97 @@ impl NetworkConfigBuilder<Initial> {
...
+            }
+        }
+
+        if !default_args.is_empty() {
+            for node in network_config.relaychain.as_mut().unwrap().nodes.iter_mut() {
+                if node.args.is_empty() {
+                    node.args.clone_from(&default_args.clone());
+                }
+            }
+        }
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:269]: We are iterating over the `nodes` here multiple times, one for each option. I think we can rewrite in the other way and iterate one time and make the check inside the loop. (I can send you a pr)

File: crates/configuration/src/shared/types.rs [hunk 6]
```diff
@@ -365,15 +365,51 @@ pub struct ValidationContext {
 
 #[derive(Default, Debug, Clone, PartialEq, Deserialize)]
 pub struct ChainDefaultContext {
-    pub(crate) default_command: Option<Command>,
```
Comment (pepoviola) [inline crates/configuration/src/shared/types.rs:287]: Hi @wirednkod, I'm not fully understand the reason behind this change. This is an internal type that is not exposed through the config builder right?

File: crates/configuration/src/shared/constants.rs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+pub const VALID_REGEX: &str = "Regex should be valid";
+pub const BORROWABLE: &str = "Must be borrowable as mutable";
+pub const RELAY_NOT_NONE: &str = "Typestate should ensure the relaychain isn't None at this point,";
+pub const SHOULD_COMPILE: &str = "Should compile with success";
+pub const INFAILABLE: &str = "Infaillible";
+pub const NO_ERR_DEF_BUILDER: &str = "Should have no errors for default builder";
+pub const RW_FAILED: &str = "Should be able to read/write - failed";
+pub const OPT_UNWRAP_ERR: &str = "Option unwrap error";
```
Comment (pepoviola) [inline crates/configuration/src/shared/constants.rs:8]: Is this used? seems like not.

File: crates/configuration/src/relaychain.rs (hunk 4 above)
Comment (pepoviola) [reply crates/configuration/src/relaychain.rs:28]:
```suggestion
    nodes: Vec<NodeConfig>,
```
we can make it private again here since you added the `set_nodes` fn

File: crates/configuration/src/shared/types.rs (hunk 6 above)
Comment (wirednkod) [reply crates/configuration/src/shared/types.rs:287]: Correct. Since this is internal and as we agreed this we will keep as `pub(crate)`;

File: crates/configuration/src/parachain.rs [hunk 8]
```diff
@@ -223,7 +229,7 @@ impl<A> ParachainConfigBuilder<A> {
             default_image: self.config.default_image.clone(),
             default_resources: self.config.default_resources.clone(),
             default_db_snapshot: self.config.default_db_snapshot.clone(),
-            default_args: self.config.default_args.clone(),
+            default_args: self.config.default_args().into_iter().cloned().collect(),
```
Comment (l0r1s) [inline crates/configuration/src/parachain.rs:232]: Not needed

```suggestion
            default_args: self.config.clone(),
```

File: crates/configuration/src/relaychain.rs [hunk 9]
```diff
@@ -78,6 +79,11 @@ impl RelaychainConfig {
     pub fn nodes(&self) -> Vec<&NodeConfig> {
         self.nodes.iter().collect::<Vec<&NodeConfig>>()
     }
+
+    /// The nodes of the relay chain.
```
Comment (l0r1s) [inline crates/configuration/src/relaychain.rs:83]: This code is not exposed publicly:

```suggestion
    // The nodes of the relay chain.
```

Given how straightforward is this code, we can even skip it ?

File: crates/configuration/src/relaychain.rs [hunk 10]
```diff
@@ -135,11 +141,11 @@ impl<A> RelaychainConfigBuilder<A> {
 
     fn default_chain_context(&self) -> ChainDefaultContext {
         ChainDefaultContext {
-            default_command: self.config.default_command.clone(),
-            default_image: self.config.default_image.clone(),
+            default_command: self.config.default_command().cloned(),
+            default_image: self.config.default_image().cloned(),
```
Comment (l0r1s) [inline crates/configuration/src/relaychain.rs:145]: Not needed, the getter are for the public API, we have access to the inner directly here:

```suggestion
            default_command: self.config.default_command.clone(),
            default_image: self.config.default_image.clone(),
```

File: crates/configuration/src/relaychain.rs [hunk 11]
```diff
@@ -135,11 +141,11 @@ impl<A> RelaychainConfigBuilder<A> {
...
             default_resources: self.config.default_resources.clone(),
             default_db_snapshot: self.config.default_db_snapshot.clone(),
-            default_args: self.config.default_args.clone(),
+            default_args: self.config.default_args().into_iter().cloned().collect(),
```
Comment (l0r1s) [inline crates/configuration/src/relaychain.rs:148]: Same

```suggestion
            default_args: self.config.default_args.clone(),
```

File: crates/configuration/src/shared/node.rs [hunk 12]
```diff
@@ -152,16 +168,30 @@ impl NodeConfig {
...
+    pub fn set_image(&mut self, image: Image) {
+        self.image = Some(image);
+    }
+
     /// Command to run the node.
     pub fn command(&self) -> Option<&Command> {
         self.command.as_ref()
     }
 
+    pub fn set_command(&mut self, command: Command) {
```
Comment (l0r1s) [inline crates/configuration/src/shared/node.rs:181]:
```suggestion
    pub(crate)  fn set_command(&mut self, command: Command) {
```

File: crates/configuration/src/shared/node.rs [hunk 13]
```diff
@@ -152,16 +168,30 @@ impl NodeConfig {
         self.image.as_ref()
     }
 
+    /// The default container image used for nodes.
+    pub fn set_image(&mut self, image: Image) {
```
Comment (l0r1s) [inline crates/configuration/src/shared/node.rs:172]: Be careful of what we expose as public API

```suggestion
    pub(crate) fn set_image(&mut self, image: Image) {
```

File: crates/configuration/src/shared/node.rs [hunk 14]
```diff
@@ -95,7 +104,14 @@ impl Serialize for NodeConfig {
...
 
-        if self.args.is_empty() || self.args == self.chain_context.default_args {
+        if self.args.is_empty()
+            || self.args()
+                == self
+                    .chain_context
+                    .default_args
+                    .iter()
+                    .collect::<Vec<&Arg>>()
+        {
```
Comment (l0r1s) [inline crates/configuration/src/shared/node.rs:114]: Why modify this code ? We don't need to use getter because we have access to self fields and PartialEq borrow automatically here: https://doc.rust-lang.org/nightly/std/cmp/trait.PartialEq.html

```suggestion
        if self.args.is_empty() || self.args == self.chain_context.default_args {
```

File: crates/configuration/src/shared/node.rs [hunk 15]
```diff
@@ -52,27 +52,36 @@ impl From<(&str, &str)> for EnvVar {
...
 
 /// A node configuration, with fine-grained configuration options.
-#[derive(Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, PartialEq, Deserialize)]
 pub struct NodeConfig {
     name: String,
-    image: Option<Image>,
-    command: Option<Command>,
+    pub(crate) image: Option<Image>,
+    pub(crate) command: Option<Command>,
```
Comment (l0r1s) [inline crates/configuration/src/shared/node.rs:59]: If we use setter we don't need to expose the fields to the crate (set_image/set_command are defined below), we can go only setters or only public field to crate to make it uniform.

File: crates/configuration/src/relaychain.rs [hunk 16]
```diff
@@ -101,7 +107,7 @@ impl Default for RelaychainConfigBuilder<Initial> {
             config: RelaychainConfig {
                 chain: "default"
                     .try_into()
-                    .expect("'default' overriding should be ensured by typestate. this is a bug, please report it: https://github.com/paritytech/zombienet-sdk/issues"),
+                    .unwrap_or_else(|_| panic!("{} {}", DEFAULT_TYPESTATE, THIS_IS_A_BUG)),
```
Comment (l0r1s) [inline crates/configuration/src/relaychain.rs:110]: We should disable this lint from clippy and revert to using expect with &format: https://github.com/rust-lang/rust-clippy/issues/2928

File: crates/configuration/src/shared/helpers.rs [hunk 17]
```diff
@@ -31,7 +32,7 @@ pub fn ensure_node_name_unique(
 ) -> Result<(), anyhow::Error> {
     let mut context = validation_context
         .try_borrow_mut()
-        .expect("must be borrowable as mutable, this is a bug please report it: https://github.com/paritytech/zombienet-sdk/issues");
+        .unwrap_or_else(|_| panic!("{}, {}", BORROWABLE, THIS_IS_A_BUG));
```
Comment (l0r1s) [inline crates/configuration/src/shared/helpers.rs:35]: Same as above

File: crates/configuration/src/shared/helpers.rs [hunk 18]
```diff
@@ -47,7 +48,7 @@ pub fn ensure_port_unique(
...
```
Comment (l0r1s) [inline crates/configuration/src/shared/helpers.rs:51]: Same as above

File: crates/configuration/src/shared/resources.rs [hunk 19]
```diff
@@ -43,7 +48,7 @@ impl TryFrom<&str> for ResourceQuantity {
     fn try_from(value: &str) -> Result<Self, Self::Error> {
         lazy_static! {
             static ref RE: Regex = Regex::new(r"^\d+(.\d+)?(m|K|M|G|T|P|E|Ki|Mi|Gi|Ti|Pi|Ei)?$")
-                .expect("should compile with success. this is a bug, please report it: https://github.com/paritytech/zombienet-sdk/issues");
+                .unwrap_or_else(|_| panic!("{}, {}", SHOULD_COMPILE, THIS_IS_A_BUG));
```
Comment (l0r1s) [inline crates/configuration/src/shared/resources.rs:51]: Same as above

File: crates/configuration/src/shared/types.rs [hunk 20]
```diff
@@ -27,6 +32,14 @@ impl From<u128> for U128 {
...
 }
 
+// impl TryFrom<&str> for U128 {
+//     type Error = ConversionError;
+
+//     fn try_from(value: &str) -> Result<Self, Self::Error> {
+//         Ok(Self(value.to_string().parse::<u128>().unwrap()))
+//     }
+// }
+
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:42]: Why is is commented ? Could probably be reused below with some modification:

```suggestion
use std::error::Error;

impl TryFrom<&str> for U128 {
     type Error = Box<dyn Error>;

     fn try_from(value: &str) -> Result<Self, Self::Error> {
         Ok(Self(value.to_string().parse::<u128>()?))
     }
 }

```

File: crates/configuration/src/shared/types.rs [hunk 21]
```diff
@@ -38,6 +51,32 @@ impl Serialize for U128 {
...
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("an integer between 0 and 2^128 − 1.")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(U128(v.to_string().parse::<u128>().unwrap()))
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:72]: With above modification we get:

```suggestion
                v.try_into().map_err(de::Error::custom)
```

File: crates/configuration/src/shared/types.rs [hunk 22]
```diff
@@ -38,6 +51,32 @@ impl Serialize for U128 {
...
+            }
+        }
+
+        deserializer.deserialize_str(U128Visitor)
+    }
+}
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:78]: I think it makes sense to put the visitor definition out of the Deserialize implementation:

```suggestion
struct U128Visitor;
        
impl<'de> de::Visitor<'de> for U128Visitor {
    type Value = U128;

    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
        formatter.write_str("an integer between 0 and 2^128 − 1.")
    }

    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
    where
        E: de::Error,
    {
        Ok(U128(v.to_string().parse::<u128>().unwrap()))
    }
}

impl<'de> Deserialize<'de> for U128 {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        deserializer.deserialize_str(U128Visitor)
    }
}
```

File: crates/configuration/src/shared/types.rs [hunk 23]
```diff
@@ -110,7 +149,7 @@ impl TryFrom<&str> for Image {
             static ref RE: Regex = Regex::new(&format!(
                 "^({IP_PART}|{HOSTNAME_PART}/)?{TAG_NAME_PART}(:{TAG_VERSION_PART})?$",
             ))
-            .expect("should compile with success. this is a bug, please report it: https://github.com/paritytech/zombienet-sdk/issues");
+            .unwrap_or_else(|_| panic!("{}, {}", SHOULD_COMPILE, THIS_IS_A_BUG));
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:152]: Same as above

File: crates/configuration/src/shared/types.rs [hunk 24]
```diff
@@ -208,7 +247,8 @@ impl From<&str> for AssetLocation {
         }
 
         Self::FilePath(
-            PathBuf::from_str(value).expect("infaillible. this is a bug, please report it"),
+            PathBuf::from_str(value)
+                .unwrap_or_else(|_| panic!("{}, {}", INFAILABLE, THIS_IS_A_BUG)),
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:251]: Same as above

File: crates/configuration/src/shared/types.rs [hunk 25]
```diff
@@ -231,6 +271,32 @@ impl Serialize for AssetLocation {
...
+            where
+                E: de::Error,
+            {
+                Ok(AssetLocation::from(v))
+            }
+        }
+
+        deserializer.deserialize_any(AssetLocationVisitor)
+    }
+}
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:298]: Same as above, remove the AssetLocationVisitor + Visitor implem trait out of the Deserialize implem

File: crates/configuration/src/shared/types.rs [hunk 26]
```diff
@@ -276,18 +342,59 @@ impl Serialize for Arg {
...
+                    }
+                } else {
+                    Ok(Arg::Flag(v.to_string()))
+                }
+            }
+        }
+
+        deserializer.deserialize_any(ArgVisitor)
+    }
+}
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:383]: Same as above

File: crates/configuration/src/shared/types.rs [hunk 27]
```diff
@@ -276,18 +342,59 @@ impl Serialize for Arg {
...
+                        let split: Vec<&str> = v.split(' ').collect::<Vec<&str>>();
+                        if split.len() == 1 {
+                            Ok(Arg::Flag(v.to_string()))
+                        } else {
+                            Ok(Arg::Option(split[0].to_string(), split[1].to_string()))
+                        }
+                    }
+                } else {
+                    Ok(Arg::Flag(v.to_string()))
+                }
```
Comment (l0r1s) [inline crates/configuration/src/shared/types.rs:377]: I think this can be more readable using 3 cases (contains '=', starts_with '--'/'-', flag) so moving inner if else out is probably better for readability, another thing is that we are indexing the split array without knowing the size of the vector, it's unsafe and can panic, imagine the case where we do "--test=".split('=')[1], we panic.

File: crates/configuration/src/utils.rs [hunk 28]
```diff
@@ -1,3 +1,7 @@
 pub fn is_true(value: &bool) -> bool {
     *value
 }
+
+pub fn default_as_true() -> bool {
+    true
+}
```
Comment (l0r1s) [inline crates/configuration/src/utils.rs:7]: Asking myself if those functions can be made ```pub(crate)``` ?

**Review by l0r1s — CHANGES_REQUESTED** [review]
Good job ! We probably should unit tests custom Serialize/Deserialize implems to be sure they work for every edge cases

File: crates/configuration/src/parachain.rs (hunk 8 above)
Comment (wirednkod) [reply crates/configuration/src/parachain.rs:232]: That does not seem correct 👀 
Why clone the config in the default args (type mismatch)

File: crates/configuration/src/shared/node.rs (hunk 15 above)
Comment (wirednkod) [reply crates/configuration/src/shared/node.rs:59]: Will remove the setters. We will keep the pub(crate)

File: crates/configuration/src/parachain.rs (hunk 8 above)
Comment (l0r1s) [reply crates/configuration/src/parachain.rs]: typo error ```self.config.default_args.clone()```

File: crates/configuration/src/utils.rs (hunk 28 above)
Comment (wirednkod) [reply crates/configuration/src/utils.rs:7]: yes they can 👍🏽

File: crates/configuration/src/shared/constants.rs [hunk 29]
```diff
@@ -0,0 +1,15 @@
...
+pub const INFAILABLE: &str = "Infaillible. ";
+pub const NO_ERR_DEF_BUILDER: &str = "Should have no errors for default builder. ";
+pub const RW_FAILED: &str = "Should be able to read/write - failed. ";
+pub const DEFAULT_TYPESTATE: &str = "'default' overriding should be ensured by typestate.";
+pub const VALIDATION_CHECK: &str = "Validation failed. ";
+
+pub const PREFIX_CANT_BE_NONE: &str = "Name prefix can't be None if a value exists.";
+
+pub const THIS_IS_A_BUG: &str =
+    "This is a bug please report it: https://github.com/paritytech/zombienet-sdk/issues";
```
Comment (l0r1s) [inline crates/configuration/src/shared/constants.rs:15]: One thing we could agree on is having non-capitalized error message ? This is something common in Rust, wdyt ?

File: crates/configuration/src/shared/resources.rs [hunk 30]
```diff
@@ -72,19 +77,19 @@ pub struct Resources {
     limit_cpu: Option<ResourceQuantity>,
 }
 
+#[derive(Serialize, Deserialize)]
+struct ResourcesField {
+    memory: Option<ResourceQuantity>,
+    cpu: Option<ResourceQuantity>,
+}
+
 impl Serialize for Resources {
```
Comment (l0r1s) [inline crates/configuration/src/shared/resources.rs:86]: :+1:

File: crates/configuration/src/shared/constants.rs (hunk 29 above)
Comment (wirednkod) [reply crates/configuration/src/shared/constants.rs:15]: I am not aware of that. If its a "thing" we should do it

File: crates/configuration/src/network.rs [hunk 31]
```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
...
+            .cloned();
+
+        let relaychain_default_image: Option<Image> = network_config
+            .relaychain
+            .as_ref()
+            .unwrap()
+            .default_image()
+            .cloned();
+
+        let relaychain_default_db_snapshot: Option<crate::shared::types::AssetLocation> =
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:97]:
```suggestion
        let relaychain_default_db_snapshot: Option<AssetLocation> =
```

this need to bring the type to the scope with 
```rs
use crate::shared::types::AssetLocation
```

File: crates/configuration/src/network.rs [hunk 32]
```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
...
+            .relaychain
+            .as_ref()
+            .unwrap()
+            .default_command()
+            .cloned();
+
+        let relaychain_default_image: Option<Image> = network_config
+            .relaychain
+            .as_ref()
+            .unwrap()
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:93]: We should annotate why is `safe` to use `unwrap` here (since we are checking `relaychain` is some before),
Also, I think this could be refactored in a nice `if let` :)

File: crates/configuration/src/network.rs [hunk 33]
```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
...
+
+        let relaychain_default_db_snapshot: Option<crate::shared::types::AssetLocation> =
+            network_config
+                .relaychain
+                .as_ref()
+                .unwrap()
+                .default_db_snapshot()
+                .cloned();
+
+        let default_args: Vec<Arg> = network_config
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:105]: We are using two ways here to access the data
`.relaychain.as_ref().unwrap()` and `.relaychain()` we should access in a uniform way here.

File: crates/configuration/src/network.rs [hunk 34]
```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
...
+        if relaychain_default_image.is_some() {
+            TryInto::<Image>::try_into(relaychain_default_image.clone().expect(VALIDATION_CHECK))?;
+        }
+        if relaychain_default_command.is_some() {
+            TryInto::<Command>::try_into(
+                relaychain_default_command.clone().expect(VALIDATION_CHECK),
+            )?;
+        }
+
+        // SAFETY: is ok to use `unwrap` here since we ensure that is some at the begging of this fn
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:130]:
```suggestion
        // SAFETY: is ok to use `unwrap` here since we ensure that is some at the begging of this fn
```
I think this comment is outdated (needs to be a few lines before :))

File: crates/configuration/src/network.rs [hunk 35]
```diff
@@ -48,11 +58,117 @@ impl NetworkConfig {
...
+
+        network_config
+            .relaychain
+            .as_mut()
+            .expect(&format!("{}, {}", NO_ERR_DEF_BUILDER, THIS_IS_A_BUG))
+            .set_nodes(nodes);
+
+        // Validation checks for parachains
+        network_config.parachains().iter().for_each(|parachain| {
+            let _ = TryInto::<Chain>::try_into(parachain.chain().unwrap().as_str());
```
Comment (pepoviola) [inline crates/configuration/src/network.rs:160]: We can use `ok_or` and return a error here instead of `unwrap` and panic.

File: crates/configuration/src/shared/constants.rs (hunk 29 above)
Comment (pepoviola) [reply crates/configuration/src/shared/constants.rs:15]: https://rust-lang.github.io/api-guidelines/interoperability.html?highlight=message#error-types-are-meaningful-and-well-behaved-c-good-err

```
The error message given by the Display representation of an error type should be lowercase without trailing punctuation, and typically concise.
```
We can take this approach :)

**Review by pepoviola — COMMENTED** [review]
Looks good @wirednkod!!! Couple of comments and nits

File: crates/configuration/src/network.rs (hunk 33 above)
Comment (wirednkod) [reply crates/configuration/src/network.rs:105]: I convert all to `.relaychain()` as it makes more sense than using `unwrap()`

File: crates/configuration/src/network.rs (hunk 32 above)
Comment (wirednkod) [reply crates/configuration/src/network.rs]: `if let` is not the best choice for this particular place you mention, as the variable is reused, and thus `cloned()` at least 2 times (and the rest ones), so I prefered to leave it as is.

**Review by pepoviola — APPROVED** [review]
